jib-cli/src/main/java/com/google/cloud/tools/jib/cli/Jar.java

```diff
@@ -198,8 +198,9 @@ public Integer call() {
       CacheDirectories cacheDirectories =
           CacheDirectories.from(commonCliOptions, jarFile.toAbsolutePath().getParent());
       JarProcessor processor = JarProcessors.from(jarFile, cacheDirectories, mode);
+      Integer jarJavaVersion = JarProcessors.getJavaMajorVersion(jarFile);
```

Comment:
Instead of making `getJavaMajorVersion()` public and calling it twice (although I have no idea if there will be noticeable performance impact when "opening a jar" twice with `new JarFile()` in this case), how about adding a field `JarProcessor.javaVersion` and a getter for it? (Maybe the current `getJavaMajorVersion()` should be renamed to, say, `{read|determine}JavaMajorVersion` if we do so.) That way, we can avoid running the code twice. I see `JarProcessor` is already passed into `toJibContainerBuilder()`, so it just needs to call the getter.

Reply:
Ah I missed this. Thanks for pointing this out - adding a getter to `JarProcessor` is a good idea. Since interfaces don't allow instance fields, I had to add a parameter to the constructors of the `JarProcessor` instances which resulted in a couple of extra changes to the test.